### ISSUE TITLE
Incorrect route is added/removed when using cf map-route/unmap-route with context path routes

### ISSUE THREAD

**Issue by d-reinhold** [cloudfoundry/cli#792]
### Command

cf map-route/unmap-route

### CLI Version

6.16.1+924508c-2016-02-26

### Error

Suppose we have the following apps and routes...

```
$ cf routes
Getting routes for org system / space test as admin ...

space   host          domain                   path   apps        service
test    shared-host   example.com			   /2     app1
test    shared-host   example.com 					  app1,app2
```

..and we want to remove the `shared-host` route with no context path:

```
$ cf unmap-route app1 example.com --hostname shared-host
Removing route shared-host.example.com/2 from app app1 in org system / space test as admin...
OK

$ cf routes
Getting routes for org system / space test as admin ...

space   host          domain                   path   apps        service
test    shared-host   example.com   		   /2
test    shared-host   example.com                     app1,app2
```

It removed the wrong route! So we run the same unmap command again:

```
$ cf unmap-route app1 example.com --hostname shared-host
Removing route shared-host.example.com/2 from app app1 in org system / space test as admin...
OK

Route to be unmapped is not currently mapped to the application.

$ cf routes
Getting routes for org system / space test as admin ...

space   host          domain                   path   apps        service
test    shared-host   example.com              /2
test    shared-host   example.com                     app1,app2
```

Now it is impossible to remove the desired route from app1.
The same issue occurs when using `cf map-route`; the incorrect route may be selected.

I believe this issue was introduced in commit https://github.com/cloudfoundry/cli/commit/9826b730b9af70a45d7f3a0438064b46fb3915ef, which was motivated by this Tracker story:
https://www.pivotaltracker.com/story/show/113911793

### CC API Version
2.43.0

### Platform Details

Mac OS X 10.10.4

### Shell

iTerm2

**Comment by dkoper** [comment]
Hi @d-reinhold 

Thanks for your report! I have reproduced the issue locally and will ask the team to investigate.

Cheers,
Dies Koper
CF CLI PM

**Comment by dkoper** [comment]
This has been fixed in cf CLI v6.17.0.
https://github.com/cloudfoundry/cli#downloads

Thanks again!
Dies Koper
CF CLI PM
